Source/JavaScriptCore/jit/JITWorklist.cpp

```diff
@@ -319,21 +302,32 @@ void JITWorklist::dump(const AbstractLocker& locker, PrintStream& out) const
         ", Num Active Threads = ", m_numberOfActiveThreads, "/", m_threads.size(), "]");
 }
 
-void JITWorklist::removeAllReadyPlansForVM(VM& vm, Vector<RefPtr<JITPlan>, 8>& myReadyPlans)
+JITWorklist::State JITWorklist::removeAllReadyPlansForVM(VM& vm, Vector<RefPtr<JITPlan>, 8>& myReadyPlans, JITCompilationKey requestedKey)
 {
     DeferGC deferGC(vm);
     Locker locker { *m_lock };
-    for (size_t i = 0; i < m_readyPlans.size(); ++i) {
-        RefPtr<JITPlan> plan = m_readyPlans[i];
+
+    bool isCompiled = false;
+    m_readyPlans.removeAllMatching([&](RefPtr<JITPlan> plan) {
         if (plan->vm() != &vm)
-            continue;
+            return false;
         if (plan->stage() != JITPlanStage::Ready)
-            continue;
-        myReadyPlans.append(plan);
-        m_readyPlans[i--] = m_readyPlans.last();
-        m_readyPlans.removeLast();
+            return false;
+        if (plan->key() == requestedKey)
+            isCompiled = true;
         m_plans.remove(plan->key());
+        myReadyPlans.append(WTFMove(plan));
+        return true;
+    });
+
+    if (!!requestedKey) {
```

Comment:
Nit: We could just have an `explicit operator bool()` on `JITCompilationKey` instead of `bool operator!` so you wouldn't need `!!` here.

Reply:
Nice, changed.